No brainer patch converted from 
	vsed -i 's/HAVE_DECL_SYS_SIGLIST/0/' common/signal.c

Index: gnupg-2.2.28/common/signal.c
===================================================================
--- gnupg-2.2.28.orig/common/signal.c
+++ gnupg-2.2.28/common/signal.c
@@ -87,7 +87,7 @@ get_signal_name( int signum )
 {
   /* Note that we can't use strsignal(), because it is not
      reentrant. */
-#if HAVE_DECL_SYS_SIGLIST && defined(NSIG)
+#if 0 && defined(NSIG)
   return (signum >= 0 && signum < NSIG) ? sys_siglist[signum] : "?";
 #else
   return NULL;
